### PR TITLE
feat(azure): skeleton for Azure Container Service quickstart

### PR DESCRIPTION
**Pull request by slack**

### CODE REVIEW

**Comment by slack** [comment]
This is in-progress, as part of the hackathon.

**Comment by slack** [comment]
I'm going to start capturing screenshots for the Azure Portal path.

**Comment by slack** [comment]
This is ready for review!

File: src/quickstart/provider/azure-acs/boot.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
+# Booting Azure Container Service
+
+## Prerequisites
+
+1. Azure Account - If you do not already have a Azure Cloud account, you can start a trial with $200 of free credit [here](https://azure.microsoft.com/en-us/free/). After completing sign up, you must add your billing information.
+2. Some form of *nix-based terminal - MacOS, Ubuntu, CentOS, Bash on Windows, etc
+<br>Where the following is present:
```
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/boot.md]: linebreaks in numbered lists? You can use the following format for a numbered list within a numbered list if that's what you'd like to achieve here:

1. foo
  a. bar
2. car
3. star

File: src/quickstart/provider/azure-acs/boot.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
+# Booting Azure Container Service
+
+## Prerequisites
+
+1. Azure Account - If you do not already have a Azure Cloud account, you can start a trial with $200 of free credit [here](https://azure.microsoft.com/en-us/free/). After completing sign up, you must add your billing information.
```
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/boot.md]: Would you mind replacing all first-person references (you, your) and write it in third person?

i.e.

> Azure Account - If an Azure Cloud Account has not been set up, trial accounts are available with $200 of free credit...

File: src/quickstart/provider/azure-acs/boot.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+### Path 1: Azure 'az' CLI
+
+1. Create an empty Azure resource group to deploy your cluster. The location of the resource group value can be changed to any datacenter.  `az account list-locations` gives the name of all locations.
+
+```
+$ export RG_NAME=myresourcegroup
+$ az resource group create --name "${RG_NAME}" --location southcentralus
+```
+
+2. Execute the command to deploy the cluster. The dns-prefix and ssh-key-value must be replaced with your own values.
```
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/boot.md]: Can you surround "dns-prefix" and "ssh-key-value" with backticks here?

File: src/quickstart/provider/azure-acs/boot.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+  --service-principal="${SP_NAME}" \
+  --client-secret="${SP_PASS}" \
+  --orchestrator-type=kubernetes --master-count=1 --agent-count=2 \
+  --agent-vm-size="Standard_D2_v2" \
+  --admin-username="k8sadmin" \
+  --name="k8sanddeis" --dns-prefix="mydnsprefix" \
+  --ssh-key-value @/home/myusername/.ssh/id_rsa.pub
+```
+
+> Note: When this is successfully executed, you'll only see this to start: `waiting for AAD role to propogate.done`.  It will take a few minutes for the cluster to complete creation.
```
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/boot.md]: We can use markdown admonitions here:

```
!!! note
    When this is successfully executed, you'll only see this to start: `waiting for AAD role to propogate.done`.  It will take a few minutes for the cluster to complete creation.
```

File: src/quickstart/provider/azure-acs/boot.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+Finally you should see something like this:
```
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/boot.md]: Finally, the output*

File: src/quickstart/provider/azure-acs/dns.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+By describing the `deis-router` service, you can see what IP address has been
+allocated by Azure Cloud for your Deis Workflow cluster:
+
+```
+$ kubectl --namespace=deis get service deis-router
+NAME          CLUSTER-IP    EXTERNAL-IP    PORT(S)                            AGE
+deis-router   10.0.60.172   13.82.148.57   80/TCP,443/TCP,2222/TCP,9090/TCP   54m
+```
+
+TODO: mention `<pending>` state
```
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/dns.md]: still need to be done?

File: src/quickstart/provider/azure-acs/dns.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+## Prepare the Hostname
+
+Now that you have the ip address of your load balancer we can use the `nip.io`
```
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/dns.md]: I don't think we should be recommending users to try out with nip.io for production deployments. nip.io and xip.io constantly go down and are just small hack projects, not intended for real production use. We should provide a link to a proper DNS service here instead like namecheap, godaddy etc. and show how the A records should be set up.

File: src/quickstart/provider/azure-acs/install-azure-acs.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+```
+$ helm repo add deis https://charts.deis.com/workflow
+```
+
+## Install Deis Workflow
+
+Now that Helm is installed and the repository has been added, install Workflow by running:
+
+```
+$ helm install deis/workflow --namespace deis --set controller.docker_tag=v2.9.0-acs,controller.org=kmala
```
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/install-azure-acs.md]: Are there changes/PRs that we need to provide in the controller for this? Optimally they should be installing the official version so we don't have to end up maintaining a fork here.

File: src/quickstart/provider/azure-acs/install-azure-acs.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+deis-builder-hy3xv            1/1       Running   5          5m
+deis-controller-g3cu8         1/1       Running   5          5m
+deis-database-rad1o           1/1       Running   0          5m
+deis-logger-fluentd-1v8uk     1/1       Running   0          5m
+deis-logger-fluentd-esm60     1/1       Running   0          5m
+deis-logger-sm8b3             1/1       Running   0          5m
+deis-minio-4ww3t              1/1       Running   0          5m
+deis-registry-asozo           1/1       Running   1          5m
+deis-router-k1ond             1/1       Running   0          5m
+deis-workflow-manager-68nu6   1/1       Running   0          5m
```
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/install-azure-acs.md]: Where is redis, the registry proxy, nsqd and the monitor?

File: src/quickstart/provider/azure-acs/boot.md (hunk 2 above)
Comment (slack) [inline src/quickstart/provider/azure-acs/boot.md]: One could do that.

File: src/quickstart/provider/azure-acs/install-azure-acs.md (hunk 8 above)
Comment (slack) [inline src/quickstart/provider/azure-acs/install-azure-acs.md]: Right, we have to have this in place until 2.10 is released with the controller fix/or acsengine and Azure ACS (UI) fix the certificate issue that this workaround requires..

File: src/quickstart/provider/azure-acs/install-azure-acs.md (hunk 9 above)
Comment (slack) [inline src/quickstart/provider/azure-acs/install-azure-acs.md]: Copied pods from the other quickstarts, think it is critical?

File: src/quickstart/provider/azure-acs/dns.md (hunk 7 above)
Comment (slack) [inline src/quickstart/provider/azure-acs/dns.md]: The goal here is a quickstart... and using nip/xip don't ring to me as a recommendation for production use.

We can have a DNS recommendation section, but that is out of scope for this PR.

Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/dns.md]: Looks like GKE also recommends this practice, so this is good for now.

**Comment by slack** [comment]
Feedback addressed, just merged some additional changes from @dtzar

File: src/quickstart/provider/azure-acs/install-azure-acs.md (hunk 9 above)
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/install-azure-acs.md]: nah we can update them all in another PR

File: src/quickstart/provider/azure-acs/install-azure-acs.md (hunk 8 above)
Comment (bacongobbler) [inline src/quickstart/provider/azure-acs/install-azure-acs.md]: Is this still relevant today? Do we want to ship this doc before v2.10 is released or do we want to give ACS  some more time to fix it on their end first?

Comment (dtzar) [inline src/quickstart/provider/azure-acs/install-azure-acs.md]: IMO we should merge now and then when the fixes come with ACS, then we can update the docs to remove those switches.  That said - there is one other thing worthy of note which we might want to include as a controller switch for this guide before if you decide to ship now: https://github.com/jimzim/deis-hackfest/blob/9019380510d84bd31cbb1fe5d5eb73594a5764eb/template/extensiondeis.sh#L26  @haroldwongms @jimzim can comment more if desired, but this helps keep persistent connection to the storage account across reboots to the master/node vms.

Comment (dtzar) [inline src/quickstart/provider/azure-acs/install-azure-acs.md]: It's a private repo, so this is what is added to the helm install -set switch in addition to the controller: "global.storage=azure,azure.accountname=$AZURE_STORAGE_ACCOUNT,azure.accountkey=$AZURE_STORAGE_ACCESS_KEY,azure.registry_container=registry,azure.database_container=database,azure.builder_container=builder"

Comment (dtzar) [inline src/quickstart/provider/azure-acs/install-azure-acs.md]: Completed [this here](https://github.com/dtzar/workflow/commit/bb54bef54ac1600d59ac1b5bc868c282cfb12b45), but now seeing a new issue with the controller pod in a crash loop.

**Comment by mboersma** [comment]
This all looks coherent. Are we still waiting for ACS fixes, or is this ready to go? Ping @slack @dtzar.

**Comment by dtzar** [comment]
Just sent another PR to @slack with some updates.  Hoping we have no crash loop after deploying a new cluster again (doing it right now).  If not, then IMO we should be good to go.

**Comment by slack** [comment]
Thanks for the PR @dtzar, I just merged your changes.

@mboersma option for the 2.10 release is to merge this but update the quickstart to use the new controller flag `K8S_API_VERIFY_TLS=false`. When the ACS certificate fixes are fully deployed later this week (or early next week) we can remove that flag from the quickstart.

**Comment by mboersma** [comment]
> When the ACS certificate fixes are fully deployed later this week (or early next week) we can remove that flag from the quickstart.

That sounds like a plan.

**Comment by bacongobbler** [comment]
merge when ready, @slack

**Comment by slack** [comment]
Dropped related issue to re-enable verification next week (https://github.com/deis/controller/issues/1193)
